### PR TITLE
add validation for notification Hub Id

### LINKED ISSUE #7669: Error http 405 when trying to import azurerm_notification_hub ressource
Hi team,

I remember successfully in the past, to import some azurerm_notification_hub ressource.
Today, I encounter an issue for import this ressource type with this error : 

> Error: Error retrieving Credentials for Notification Hub "" (Namespace "notifications-xxx-yyy" / Resource Group "RG-XXX-NOTIF"): notificationhubs.Client#GetPnsCredentials: Failure responding to request: StatusCode=405 -- Original Error: autorest/azure: Service returned an error. Status=405 Code="HttpMethodIsNotSupported" Message="The HTTP method 'POST' is not supported."

I tried on multiple ressources, in different subscriptions, but still have this issue and can't import anymore some notification hub ressources.

Thanks for your help, for check if I'm alone in this case or not.
Regars
Alexandre

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version
Terraform 12.26
Azure RM Provider 2.17

### Affected Resource(s)

* `azurerm_notification_hub ressource`

### Terraform Configuration Files

```hcl
resource "azurerm_notification_hub" "nh-xxx" {
  name                = lower("nh-${var.environment}-xxx")
  namespace_name      = azurerm_notification_hub_namespace.nh_namespace.name
  resource_group_name = azurerm_resource_group.rgproject.name
  location            = azurerm_resource_group.rgproject.location

  apns_credential {
    application_mode = "Production"
    bundle_id        = "com.xxx.yyy"
    key_id           = "ABCDEF"
    team_id          = "GHIJKL"
    token            = "MIGT...CNM"
  }
  gcm_credential {
    api_key = "AA...Un73"
  }

  tags                = local.common_tags
}
```

### Debug Output

azurerm_notification_hub.nh-xxx: Importing from ID "/subscriptions/xxx/resourceGroups/RG-xxx-NOTIF/providers/Microsoft.NotificationHubs/namespaces/notifications-xxx/NotificationHubs/notif-xxx-hub"...
azurerm_notification_hub.nh-xxx: Import prepared!
  Prepared azurerm_notification_hub for import
azurerm_notification_hub.nh-xxx: Refreshing state... [id=/subscriptions/3xxx/resourceGroups/RG-xxx-NOTIF/providers/Microsoft.NotificationHubs/namespaces/notifications-xxx-xxx/NotificationHubs/notif-xxx-hub]

Error: Error retrieving Credentials for Notification Hub "" (Namespace "notifications-xxx-xxxr" / Resource Group "RG-xxx-NOTIF"): notificationhubs.Client#GetPnsCredentials: Failure responding to request: StatusCode=405 -- Original Error: autorest/azure: Service returned an error. Status=405 Code="HttpMethodIsNotSupported" Message="The HTTP method 'POST' is not supported."

### Expected Behavior
Expected that Terraform import in state the ressource successfully

### Actual Behavior
The ressource is not imported

### Steps to Reproduce
1. Create config block azurerm_notification_hub
2. `terraform import -var-file="./Env/$target_env.tf-variables.tfvars" azurerm_notification_hub.nh-xxxe "/subscriptions/xxx/resourceGroups/RG-xxx-NOTIF/providers/Microsoft.NotificationHubs/namespaces/notifications-xxx-xxx/NotificationHubs/notif-xxx-hub"`

* #0000

### PR DESCRIPTION
**Pull request by njuCZ**
fix #7669

### CODE REVIEW

File: azurerm/internal/services/notificationhub/notification_hub_resource.go [hunk 1]
```diff
@@ -231,33 +232,30 @@ func resourceArmNotificationHubRead(d *schema.ResourceData, meta interface{}) er
 	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
-	id, err := azure.ParseAzureResourceID(d.Id())
+	id, err := parse.NotificationHubID(d.Id())
```
Comment (tombuildsstuff) [inline azurerm/internal/services/notificationhub/notification_hub_resource.go:235]: can we add this function to the Import check too, which should raise this at that point ([example](https://github.com/terraform-providers/terraform-provider-azurerm/blob/5785c873ab5d13043b91a71d5904d9612af338c0/azurerm/internal/services/analysisservices/analysis_services_server_resource.go#L40-L43))

File: azurerm/internal/services/notificationhub/parse/notification_hub.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+	app := NotificationHubId{
+		ResourceGroup: id.ResourceGroup,
+	}
+
+	if app.NamespaceName, err = id.PopSegment("namespaces"); err != nil {
+		return nil, fmt.Errorf("parsing Notification Hub ID %q: %+v", input, err)
+	}
+
+	if app.Name, err = id.PopSegment("notificationHubs"); err != nil {
+		return nil, fmt.Errorf("parsing Notification Hub ID %q: %+v", input, err)
```
Comment (tombuildsstuff) [inline azurerm/internal/services/notificationhub/parse/notification_hub.go:30]: this should already be wrapped:

```suggestion
		return nil, err
```

File: azurerm/internal/services/notificationhub/parse/notification_hub.go [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+	if err != nil {
+		return nil, fmt.Errorf("parsing Notification Hub ID %q: %+v", input, err)
+	}
+
+	app := NotificationHubId{
+		ResourceGroup: id.ResourceGroup,
+	}
+
+	if app.NamespaceName, err = id.PopSegment("namespaces"); err != nil {
+		return nil, fmt.Errorf("parsing Notification Hub ID %q: %+v", input, err)
```
Comment (tombuildsstuff) [inline azurerm/internal/services/notificationhub/parse/notification_hub.go:26]: this should already be wrapped:

```suggestion
		return nil, err
```

**Comment by njuCZ** [comment]
@tombuildsstuff thanks for the suggestions. I have made an update. Please have a look again.

**Pull request merged**